### DISCUSSION TITLE
Add an option for GLTF to consider Collections over Empties for parenting

### DISCUSSION THREAD

**Discussion by bertodelrio256** [godotengine/godot-proposals#7013] [Editor]
In the current configuration, the godot GLTF importer looks for empties to parent objects to spatial nodes. Id like to have the option to use blender Collections instead. Collections are very easy to work with in blender/ Having to constantly parent and unparent to empties in blender is a pain. its also like a 4 step process moving objects to other empties. With Collections, you can also easily select all objects within them.

So the idea would be, the if you have a collection named foliage then the import would put all objects in that collection under a spatial node named foliage.

**Comment by bertodelrio256** [comment]
i doubt anyone will even read this.

**Reply by dsnopek** [reply]
If you mean your proposal, yes, people will read it :-) I'm subscribed to all the activity on godot-proposals and read most of them in my e-mail. The importer isn't my area, though, so I don't personally have any thing to add here

**Reply by bertodelrio256** [reply]
Thanks! I realize after posting this that the Blender GLTF exporter obviously does not export collection information so this proposal is not possible.

**Comment by lyuma** [comment]
This sounds like a missing feature in Blender, not something Godot can handle. Collection names are not saved into the gltf file, so it needs to happen before export.

See here for the Blender issue: KhronosGroup/glTF-Blender-IO#1456

In terms of workflow, I tried using a workflow with parenting to empties in Blender, and it is very difficult to ensure that parenting is done properly, ensuring that the `Parent Inverse` is cleared or the parent transform remains at 0,0,0 while parenting the child. So I can see the value in this type of design, but it depends on Blender, and it has ambiguities when armatures are involved.
